src/asm/symbol.c

```diff
@@ -276,6 +276,13 @@ static bool isReferenced(struct Symbol const *sym)
 	return sym->ID != (uint32_t)-1;
 }
 
+struct SymbolListNode {
+	struct Symbol *sym;
+	struct SymbolListNode *next;
```

Comment:
I know RGBASM has a lot of those, mostly because it was the path of least resistance and we haven't fixed them, but I'd prefer at least all newly-introduced code to use arrays instead of linked lists.

Linked lists require more allocations, and are generally slower to traverse; they're good at insertion and deletion of items at arbitrary points (and not much else), but we're not doing either of those here.